libs/providers/vmware.py

```
@@ -1336,21 +1337,24 @@ def clone_vm(
+
                         device_spec = vim.vm.device.VirtualDeviceSpec()
                         device_spec.operation = vim.vm.device.VirtualDeviceSpec.Operation.edit
                         device_spec.device = device
-                        device_spec.device.addressType = "generated"
                         device_changes.append(device_spec)
-                        LOGGER.info(
-                            f"Configured MAC regeneration for network device: {
-                                device.deviceInfo.label if device.deviceInfo else 'Unknown'
-                            }",
-                        )
+                        LOGGER.info(f"Configured MAC regeneration for network device: {nic_label}")
```

Ruff (0.15.17)

[warning] Logging statement uses f-string

(G004)